test/utils.tests.js

```diff
@@ -0,0 +1,49 @@
+const assert = require('chai').assert;
+
+const { extractERLQuota } = require('../lib/utils');
+const { set, reset } = require('mockdate');
+
+describe('utils', () => {
+  describe('extractERLQuota', () => {
+    const dateTests = [
+      {
+        date: '2024-03-15T12:00:00.000Z',
+        expiration: 1711929600000,
+        name: '16 days, 12 hs left to end of month'
+      },
+      {
+        date: '2024-03-31T23:00:00.000Z',
+        expiration: 1711929600000,
+        name: '1 hour left to end of month'
+      },
+      {
+        date: '2024-03-31T23:59:59.000Z',
+        expiration: 1711929600000,
+        name: '1 second left to end of month'
+      }
+    ];
+
+    dateTests.forEach(test => {
+      it(`should return appropriate key, amount., and expiration when there's ${test.name}`, () => {
+        set(test.date);
+
+        const params = {
+          erlQuota: {
+            key: 'erlQuotaKey',
+            per_cal_month: 192
```

Comment:
I don't like the `*per_month` because it can mean a month from the moment it starts counting. That's why I'm doing it super specific adding the "cal" from "calendar" into the name. 
Regarding the `per_cal_month` possibly being used to track something else than quota, it's totally possible cause it's not intended just for quota, it's a variation of the `per_interval` values.
I think it also makes sense following the same convention as for the filling interval for the buckets where it's `per_second`, `per_minute`, etc, and not `tokens_per_second`.

Reply:
What I think is `tokens_per_second` would have been a better convention than the current `per_minute`, `per_second` etc... but I this PR is not the place to do such a big scale change, I also understand that. 😢

Reply:
Also (again just a nit) but why not `per_calendar_month` explicitly? Genuinely curious, just not a fan of shortenings unless we have storage concerns

(meanwhile consider this thread resolved 👍 got your point and nothing blocking here)

Reply:
that's a good suggestion. Not sure why I shortened it :D. I'll fix that